pyo3-macros-backend/src/pyclass.rs

```diff
@@ -1193,6 +1294,73 @@ fn complex_enum_struct_variant_new<'a>(
     crate::pymethod::impl_py_method_def_new(&variant_cls_type, &spec, ctx)
 }
 
+fn complex_enum_tuple_variant_new<'a>(
+    cls: &'a syn::Ident,
+    variant: &'a PyClassEnumTupleVariant<'a>,
+    ctx: &Ctx,
+) -> Result<MethodAndSlotDef> {
+    let Ctx { pyo3_path } = ctx;
+
+    let variant_cls: Ident = format_ident!("{}_{}", cls, variant.ident);
+    let variant_cls_type: syn::Type = parse_quote!(#variant_cls);
+
+    let arg_py_ident: syn::Ident = parse_quote!(py);
+    let arg_py_type: syn::Type = parse_quote!(#pyo3_path::Python<'_>);
+
+    let args = {
+        let mut no_pyo3_attrs = vec![];
+        let attrs = crate::pyfunction::PyFunctionArgPyO3Attributes::from_attrs(&mut no_pyo3_attrs)?;
+
+        let mut args = vec![
+            // py: Python<'_>
+            FnArg {
+                name: &arg_py_ident,
+                ty: &arg_py_type,
+                optional: None,
+                default: None,
+                py: true,
+                attrs: attrs.clone(),
+                is_varargs: false,
+                is_kwargs: false,
+                is_cancel_handle: false,
+            },
+        ];
+
+        for (i, field) in variant.fields.iter().enumerate() {
+            // ! Warning : This leaks memory. I have no idea how else to do this - is this even bad?
```

Comment:
There's no better way with the current `FnArg`, but of course it could be changed to contain a `Ident` instead of `&Ident`, which would necessitate a clone in other places.

Reply:
Or, better, a `Cow<Ident>`.

Reply:
I switched this in #4157.